libmscore/hairpin.cpp

```diff
@@ -231,20 +237,27 @@ void HairpinSegment::layout()
       if (autoplace()) {
             qreal minDistance = spatium() * .7;
             qreal ymax = pos().y();
+            qreal ddiff = spatium() * 0.5;
 
             SkylineLine sl(!hairpin()->placeAbove());
             sl.add(shape().translated(pos()));
             if (hairpin()->placeAbove()) {
+                  ymax = qMin(ymax, dymax - ddiff);
                   qreal d  = system()->topDistance(staffIdx(), sl);
                   if (d > -minDistance)
                         ymax -= d + minDistance;
                   }
             else {
+                  ymax = qMax(ymax, dymax - ddiff);
                   qreal d  = system()->bottomDistance(staffIdx(), sl);
                   if (d > -minDistance)
                         ymax += d + minDistance;
                   }
             rypos() += ymax - pos().y();
+            if (sd && sd->autoplace() && sd->placement() == hairpin()->placement())
+                  sd->rypos() = rypos() + ddiff + offset().y() - sd->offset().y();
+            if (ed && ed->autoplace() && ed->placement() == hairpin()->placement())
+                  ed->rypos() = rypos() + ddiff + offset().y() - ed->offset().y();
```

Comment:
Just a minor note:
`ed->rypos() = rypos() + ...`
can be written as
`ed->rypos() += ...`
which is a bit shorter and (maybe) is a bit better to read. At least it is done so in other places in the code.

Reply:
not the same, though - ed->rypos() is not rypos() (ed is end dynamic, "this" is the hairpin)

Reply:
Ah, didn't notice that, sorry.